src/MusicTheory/ScaleClass.elm

```diff
@@ -0,0 +1,165 @@
+module MusicTheory.ScaleClass exposing
+    ( HeptatonicScaleClassIntervals
+    , HexatonicScaleClassIntervals
+    , OctatonicScaleClassIntervals
+    , PentatonicScaleClassIntervals
+    , ScaleClass(..)
```

Comment:
We should make this opaque and define some constructors.

Reply:
Also, I am all in favor of making types opaque and limiting the public API, but I think we don't need to be so strict about it just yet. In some cases, like this one, modules may need to expose some internals to each other. But we have a "last line of defense" in the `elm.json` file's list of `exposed-modules`, where we can define which modules get exposed to callers outside. If I remember correctly, sometimes package authors will define "wrapper" modules that only expose what they want exposed, and not include the other two in `exposed-modules`. I can try to find some examples of that.

Reply:
I think `ScaleClass` should be exposed, at least at some point.

Clients might want to define their own scale classes. There should be dedicated constructors that validate the input and return `Result` or something similar.

But this could be for the future.